napari/_qt/layer_controls/qt_labels_controls.py

```diff
@@ -403,7 +405,9 @@ def change_preserve_labels(self, state):
         state : QCheckBox
             Checkbox indicating if overwriting label is enabled.
         """
-        self.layer.preserve_labels = state == Qt.CheckState.Checked
+        self.layer.preserve_labels = (
+            Qt.CheckState(state) == Qt.CheckState.Checked
+        )
```

Comment:
That won't work, the problem is on the right hand side.
Qt.CheckState.Checked is an enum so the comparison still fails.
This can work:
`        self.layer.contiguous = state == int(Qt.CheckState.Checked)`
but I'm not sure I like it, because it seems more likely to break again in the future.

Reply:
Actually no, this `self.layer.contiguous = state == int(Qt.CheckState.Checked)` doesn't work:
`TypeError: int() argument must be a string, a bytes-like object or a real number`

Reply:
Maybe: `self.layer.contiguous = state == Qt.CheckState.Checked.value` would work?
But still I think I prefer the more explicit comparison. 🤷‍♂️

Reply:
I also prefer the more explicit comparison (mostly because I don't need to go look up the enum values to understand the code). But I wouldn't oppose the cast to `bool` suggested because Qt's check state is quite well known - wouldn't `self.layer.preserve_labels = bool(state)` work?